### PR TITLE
Razor Class Library template supports RP/MVC or components

### LINKED ISSUE #16189: RCL docs should explain information about components
[EDIT by guardrex to update the metadata]

The RCL doc says a lot about Razor Pages and Razor Views for MVC.

It doesn't say a whole lot about adding components. We should have a section here that explains that you can use RCL for components, and what exactly is required (project file, `@using Microsoft.AspNetCore.Components.Web`). This is feedback from https://github.com/aspnet/AspNetCore/issues/17809

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 37b02665-c8ab-bd91-8057-3b07b4e238d3
* Version Independent ID: e4f176f5-0b94-d532-53d5-cffce437961a
* Content: [Razor components class libraries](https://docs.microsoft.com/en-us/aspnet/core/blazor/class-libraries?view=aspnetcore-3.0&tabs=visual-studio)
* Content Source: [aspnetcore/blazor/class-libraries.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/class-libraries.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #16189

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/class-libraries?view=aspnetcore-3.1&branch=pr-en-us-16684)

* The issue with the check box/CLI switch to enable the RCL for pages+views is a point-in-time problem. I think we just need to call it out where we describe generating the lib.
* I can't version this (yet). Engineering issue https://github.com/dotnet/aspnetcore/issues/17870 isn't triaged for a particular release.
* I also cross-link our Blazor Component Lib topic from the Razor Class Lib topic.

### CODE REVIEW

File: aspnetcore/blazor/class-libraries.md [hunk 1]
```diff
@@ -39,6 +39,17 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
    1. Right-click the app project. Select **Add** > **Reference**.
    1. Select the RCL project. Select **OK**.
 
+> [!NOTE]
+> If the **Support pages and views** check box is selected when generating the RCL from the template, the generated project doesn't include the *_Imports.razor* file with the following contents:
```
Comment (danroth27) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
> If the **Support pages and views** check box is selected when generating the RCL from the template, then also add a *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
```

File: aspnetcore/blazor/class-libraries.md [hunk 2]
```diff
@@ -47,6 +58,18 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
    dotnet new razorclasslib -o MyComponentLib1
    ```
 
+   > [!NOTE]
+   > If the `-s|--support-pages-and-views` switch is used when generating the RCL from the template, the generated project doesn't include the *_Imports.razor* file with the following contents:
```
Comment (danroth27) [inline aspnetcore/blazor/class-libraries.md]: Same suggested updates as above

File: aspnetcore/blazor/class-libraries.md [hunk 3]
```diff
@@ -39,6 +39,17 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
...
+> [!NOTE]
+> If the **Support pages and views** check box is selected when generating the RCL from the template, the generated project doesn't include the *_Imports.razor* file with the following contents:
+>
+> ```razor
+> @using Microsoft.AspNetCore.Components.Web
+> ```
+>
+> Manually add the file the root of the generated project.
+>
+> This issue will be addressed in a future release of ASP.NET Core. For more information, see [RCL template either supports MVC OR supports Components (dotnet/aspnetcore #17870)](https://github.com/dotnet/aspnetcore/issues/17870).
```
Comment (danroth27) [inline aspnetcore/blazor/class-libraries.md]: I don't think we need to link to the issue here in the docs. This isn't a blocking bug. Let's remove this line.

File: aspnetcore/blazor/class-libraries.md [hunk 4]
```diff
@@ -39,6 +39,17 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
...
 
+> [!NOTE]
+> If the **Support pages and views** check box is selected when generating the RCL from the template, the generated project doesn't include the *_Imports.razor* file with the following contents:
+>
+> ```razor
+> @using Microsoft.AspNetCore.Components.Web
+> ```
+>
+> Manually add the file the root of the generated project.
+>
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
```

File: aspnetcore/blazor/class-libraries.md (hunk 3 above)
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
```

File: aspnetcore/blazor/class-libraries.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/blazor/class-libraries.md]:
```suggestion
> If the `-s|--support-pages-and-views` switch is used when generating the RCL from the template, then also add a *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
```

File: aspnetcore/blazor/class-libraries.md [hunk 5]
```diff
@@ -47,6 +56,18 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
...
 
+   > [!NOTE]
+   > If the `-s|--support-pages-and-views` switch is used when generating the RCL from the template, then also add a *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
+   >
+   > ```razor
+   > @using Microsoft.AspNetCore.Components.Web
+   > ```
+   >
+   > Manually add the file the root of the generated project.
+   >
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
```

File: aspnetcore/blazor/class-libraries.md [hunk 6]
```diff
@@ -47,6 +56,18 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
...
+   > This issue will be addressed in a future release of ASP.NET Core. For more information, see [RCL template either supports MVC OR supports Components (dotnet/aspnetcore #17870)](https://github.com/dotnet/aspnetcore/issues/17870).
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
```

File: aspnetcore/blazor/class-libraries.md [hunk 7]
```diff
@@ -47,6 +56,18 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
...
+
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
```

File: aspnetcore/blazor/class-libraries.md [hunk 8]
```diff
@@ -47,6 +56,15 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
    dotnet new razorclasslib -o MyComponentLib1
    ```
 
+   > [!NOTE]
+   > If the `-s|--support-pages-and-views` switch is used when generating the RCL from the template, then also add a *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
> If the `-s|--support-pages-and-views` switch is used when generating the RCL from the template, then also add an *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
```

File: aspnetcore/blazor/class-libraries.md [hunk 9]
```diff
@@ -39,6 +39,15 @@ Follow the guidance in the <xref:blazor/get-started> article to configure your e
    1. Right-click the app project. Select **Add** > **Reference**.
    1. Select the RCL project. Select **OK**.
 
+> [!NOTE]
+> If the **Support pages and views** check box is selected when generating the RCL from the template, then also add a *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
```
Comment (guardrex) [inline aspnetcore/blazor/class-libraries.md]:
```suggestion
> If the **Support pages and views** check box is selected when generating the RCL from the template, then also add an *_Imports.razor* file to root of the generated project with the following contents to enable Razor component authoring:
```

**Comment by guardrex** [comment]
@danroth27 *Done* :+1:

**Comment by guardrex** [comment]
@danroth27 Did you want to take a last look? :eyes:

**Pull request merged**